src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultTagHelperDescriptorFactory.cs

```diff
@@ -719,14 +882,34 @@ private bool IsAccessibleProperty(IPropertySymbol property)
         /// ONE1TWO2THREE3 => one1two2three3
         /// First_Second_ThirdHi => first_second_third-hi
         /// </example>
-        private static string ToHtmlCase(string name)
+        internal static string ToHtmlCase(string name)
         {
             return HtmlCaseRegex.Replace(name, HtmlCaseRegexReplacement).ToLowerInvariant();
         }
 
         private static string GetFullName(ITypeSymbol type)
         {
-            return type.ContainingNamespace.ToDisplayString() + "." + type.Name;
+            switch (type.SpecialType)
+            {
+                case SpecialType.System_Object:
+                case SpecialType.System_Boolean:
+                case SpecialType.System_Char:
+                case SpecialType.System_SByte:
+                case SpecialType.System_Byte:
+                case SpecialType.System_Int16:
+                case SpecialType.System_UInt16:
+                case SpecialType.System_Int32:
+                case SpecialType.System_UInt32:
+                case SpecialType.System_Int64:
+                case SpecialType.System_UInt64:
+                case SpecialType.System_Decimal:
+                case SpecialType.System_Single:
+                case SpecialType.System_Double:
+                case SpecialType.System_String:
```

Comment:
`ToDisplayString()` has `string`, `long` etc. stay as `string`, `long`. In the old variant of the descriptor factories all types always had their full names returned. I couldn't find any `IsPrimitive` equivalent on the `ITypeSymbol` so feel free to suggest if you know something better.